File: 0011_Container With Most Water/ContainerWithMostWater.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        if not height or len(height) <= 1:return 0
        left = 0
        right = len(height) - 1
        ans = 0
        while left < right:
            ans = max(ans,min(height[left], height[right]) * (right - left))
            if height[left] > height[right]:
                right -= 1
            else:
                left += 1
        return ans

File: 0011_Container With Most Water/ContainerWithMostWater.cs
public class Solution {
    public int MaxArea (int[] height) {
        if (height == null || height.Length <= 1) return 0;
        int left = 0;
        int right = height.Length - 1;
        int max = 0;
        while (left < right) {
            max = Math.Max (max, Math.Min (height[left], height[right]) * (right - left));
            if (height[left] < height[right]) {
                left++;
            } else {
                right--;
            }
        }

        return max;
    }
}